t/unit/app/test_loaders.py

```diff
@@ -234,19 +234,74 @@ def test_autodiscover_tasks(self):
             base.autodiscover_tasks(['foo'])
             frm.assert_called()
 
-    def test_find_related_module(self):
+    # Happy - get something back
+    def test_find_related_module__when_existent_package_alone(self):
         with patch('importlib.import_module') as imp:
             imp.return_value = Mock()
             imp.return_value.__path__ = 'foo'
-            assert base.find_related_module('bar', 'tasks').__path__ == 'foo'
-            imp.assert_any_call('bar')
-            imp.assert_any_call('bar.tasks')
+            assert base.find_related_module('foo', None).__path__ == 'foo'
+            imp.assert_called_once_with('foo')
 
-            imp.reset_mock()
-            assert base.find_related_module('bar', None).__path__ == 'foo'
-            imp.assert_called_once_with('bar')
+    def test_find_related_module__when_existent_package_and_related_name(self):
+        with patch('importlib.import_module') as imp:
+            first_import = Mock()
+            first_import.__path__ = 'foo'
+            second_import = Mock()
+            second_import.__path__ = 'foo/tasks'
+            imp.side_effect = [first_import, second_import]
+            assert base.find_related_module('foo', 'tasks').__path__ == 'foo/tasks'
+            imp.assert_any_call('foo')
+            imp.assert_any_call('foo.tasks')
+
+    def test_find_related_module__when_existent_package_parent_and_related_name(self):
+        with patch('importlib.import_module') as imp:
+            first_import = ModuleNotFoundError(name='foo.BarApp')  # Ref issue #2248
```

Comment:
It seems like this behavior was added to match some edge case w/ how Django apps present themselves for auto-discovery (see #2248).